File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppJava/app/src/main/java/pl/udu/uwr/pum/verynobleappjava/data/laureateresponse/LaureateResponseItem.java
package pl.udu.uwr.pum.verynobleappjava.data.laureateresponse;

import java.util.ArrayList;

import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.laureate.Birth;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.laureate.Death;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.laureate.FamilyName;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.laureate.FullName;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.laureate.GivenName;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.laureate.KnownName;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.laureate.Link;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.meta.Meta;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.nobelprize.NobelPrize;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.wiki.Wikidata;
import pl.udu.uwr.pum.verynobleappjava.data.laureateresponse.wiki.Wikipedia;

public class LaureateResponseItem {
    public String id;
    public KnownName knownName;
    public GivenName givenName;
    public FamilyName familyName;
    public FullName fullName;
    public String fileName;
    public String gender;
    public Birth birth;
    public Death death;
    public Wikipedia wikipedia;
    public Wikidata wikidata;
    public ArrayList<String> sameAs;
    public ArrayList<Link> links;
    public ArrayList<NobelPrize> nobelPrizes;
    public Meta meta;
}

File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppKotlin/app/src/main/java/pl/udu/uwr/pum/verynobleappkotlin/data/laureateresponse/LaureateResponseItem.kt
package pl.udu.uwr.pum.verynobleappkotlin.data.laureateresponse

import pl.udu.uwr.pum.verynobleappkotlin.data.laureateresponse.laureate.*
import pl.udu.uwr.pum.verynobleappkotlin.data.laureateresponse.meta.Meta
import pl.udu.uwr.pum.verynobleappkotlin.data.laureateresponse.nobelprize.NobelPrize
import pl.udu.uwr.pum.verynobleappkotlin.data.laureateresponse.wiki.Wikidata
import pl.udu.uwr.pum.verynobleappkotlin.data.laureateresponse.wiki.Wikipedia

data class LaureateResponseItem(
    val birth: Birth,
    val death: Death?,
    val familyName: FamilyName,
    val fileName: String,
    val fullName: FullName,
    val gender: String,
    val givenName: GivenName,
    val id: String,
    val knownName: KnownName,
    val links: List<Link>,
    val meta: Meta,
    val nobelPrizes: List<NobelPrize>,
    val sameAs: List<String>,
    val wikidata: Wikidata,
    val wikipedia: Wikipedia
)